pkg/asset/installconfig/powervs/session.go

```diff
@@ -67,28 +58,31 @@ type SessionVars struct {
 	PowerVSResourceGroup string
 }
 
-func authenticateAPIKey(sess *bxsession.Session) error {
-	config := sess.Config
-	tokenRefresher, err := authentication.NewIAMAuthRepository(config, &rest.Client{
-		DefaultHeader: gohttp.Header{
-			"User-Agent": []string{http.UserAgent()},
-		},
-	})
+func authenticateAPIKey(apikey string) (string, error) {
+	a, err := core.NewIamAuthenticatorBuilder().SetApiKey(apikey).Build()
 	if err != nil {
-		return err
+		return "", err
 	}
-	return tokenRefresher.AuthenticateAPIKey(config.BluemixAPIKey)
+	token, err := a.GetToken()
+	if err != nil {
+		return "", err
+	}
+	return token, nil
 }
 
-func fetchUserDetails(sess *bxsession.Session) (*User, error) {
-	config := sess.Config
+func fetchUserDetails(apikey string) (*User, error) {
 	user := User{}
 	var bluemixToken string
 
-	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
-		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
+	iamToken, err := authenticateAPIKey(apikey)
+	if err != nil {
+		return &user, err
+	}
+
+	if strings.HasPrefix(iamToken, "Bearer") {
+		bluemixToken = iamToken[7:]
```

Comment:
Maybe a comment explaining where the magic "7" comes from?

Reply:
Good catch, replacing with `len("Bearer ")`...

Reply:
Or the second member of https://pkg.go.dev/strings#Fields

Reply:
Force-pushed. PTAL